ddtrace/internal/_file_queue.py

```diff
@@ -0,0 +1,63 @@
+import os
+import os.path
+import tempfile
+import typing
+
+from ddtrace.internal._unpatched import unpatched_open
+
+
+try:
+    # Unix based file locking
+    # Availability: Unix, not Emscripten, not WASI.
+    import fcntl
+
+    def lock(f):
+        fcntl.lockf(f, fcntl.LOCK_EX)
+
+    def unlock(f):
+        fcntl.lockf(f, fcntl.LOCK_UN)
+
+except ModuleNotFoundError:
+    # Availability: Windows
+    import msvcrt
+
+    def size(f):
+        return os.path.getsize(os.path.realpath(f.name))
+
+    def lock(f):
+        msvcrt.locking(f.fileno(), msvcrt.LK_RLCK, size(f))
+
+    def unlock(f):
+        msvcrt.locking(f.fileno(), msvcrt.LK_UNLCK, size(f))
+
+
+class File_Queue:
+    """A simple file-based queue implementation for multiprocess communication."""
+
+    def __init__(self) -> None:
+        self.directory = tempfile.gettempdir()
+        self.filename = os.path.join(self.directory, "file_queue")
+
+    def put(self, data: str) -> None:
+        """Push a string to the queue."""
+        try:
+            with unpatched_open(self.filename, "a") as f:
+                lock(f)
+                f.write(data + "\x00")
```

Comment:
It would be nice to write only unique data, to avoid the file from getting too big? Alternatively, limit the size it can get, no?

Reply:
the main thread should delete the file every 5 seconds. I could add a failsafe that would check the file size and not write if too large, but it would cost a little time. not sure right now if it's better or worse.